specification/security/resource-manager/Microsoft.Security/preview/2020-08-06-preview/onPremiseIotSensors.json

```diff
@@ -0,0 +1,276 @@
+        "operationId": "OnPremiseIotSensors_List",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionId"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/OnPremiseIotSensorsList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Security/onPremiseIotSensors/{onPremiseIotSensorName}": {
+      "get": {
+        "x-ms-examples": {
+          "Get on-premise IoT sensor": {
+            "$ref": "./examples/OnPremiseIotSensors/Get.json"
+          }
+        },
+        "tags": [
+          "On-Premise IoT Sensors"
+        ],
+        "operationId": "OnPremiseIotSensors_Get",
+        "description": "Get on-premise IoT sensor",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "#/parameters/OnPremiseIotSensorName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/OnPremiseIotSensor"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../common/v1/types.json#/definitions/CloudError"
+            }
+          }
+        }
+      },
+      "put": {
+        "x-ms-examples": {
+          "Create or update on-premise IoT sensor": {
+            "$ref": "./examples/OnPremiseIotSensors/Put.json"
+          }
+        },
+        "tags": [
+          "On-Premise IoT Sensors"
+        ],
+        "operationId": "OnPremiseIotSensors_CreateOrUpdate",
+        "description": "Create or update on-premise IoT sensor",
+        "parameters": [
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersion"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionId"
+          },
+          {
+            "$ref": "#/parameters/OnPremiseIotSensorName"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Updated on-premise IoT sensor",
+            "schema": {
+              "$ref": "#/definitions/OnPremiseIotSensor"
+            }
+          },
+          "201": {
```

Comment:
Is this long-running operation?

Reply:
No, it's a synchronous operation that completes immediately.